osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -618,27 +617,32 @@ protected override void Update()
                     // Add those items within the previously found index range that should be displayed.
                     foreach (var item in toDisplay)
                     {
-                        var panel = setPool.Get(p => p.Item = item);
+                        var panel = setPool.Get();
 
+                        panel.Item = item;
                         panel.Depth = item.CarouselYPosition;
-                        panel.Y = item.CarouselYPosition;
+                        panel.UpdateY();
 
                         Scroll.Add(panel);
                     }
                 }
             }
 
+            // Update Y positions for the non-drawable models
+            if (Clock != null)
+            {
+                foreach (var set in beatmapSets)
```

Comment:
is this not updated *every* panel ever? this is going to be a performance issue (the very reason everything else in this whole class is structured to specifically avoid this).

Reply:
I have 5392 beatmaps and see no difference in performance compared to master.

Reply:
How did you measure that?

Reply:
Just looking at FPS. I get ~100 FPS update when scrolling in both this PR and current release

Reply:
A more accurate (and believable) way to do that would be to either profile or run a benchmark.